test/unit_test/static/unit_constants.cpp

```diff
@@ -0,0 +1,66 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include "units/physical/dimensions/length.h"
+#include "units/physical/dimensions/speed.h"
+#include "units/physical/si/base/length.h"
+#include "units/physical/si/base/mass.h"
+#include "units/physical/si/base/time.h"
+#include "units/physical/si/derived/area.h"
+#include "units/physical/si/derived/speed.h"
+#include <type_traits>
+
+using namespace units::physical;
+using namespace units::physical::si;
+
+constexpr auto m_per_s = m / s;
+
+static_assert(std::is_same_v<decltype(m_per_s)::rep, decltype(m)::rep>);
+
+static_assert(2 * m == 2_q_m);
+static_assert(2 * s == 2_q_s);
+
+static_assert(2_q_m / s == 2_q_m_per_s);
+static_assert(2 * m / s == 2_q_m_per_s);
+static_assert(2 / s * m == 2_q_m_per_s);
+static_assert(2 * (m / s) == 2_q_m_per_s);
+static_assert(2 * m_per_s == 2_q_m_per_s);
+
```

Comment:
Could you please add some negative tests checking that invalid expressions do not compile? With the `one_rep` implementation I assume that `s / 2` is an invalid expression, right? Should it be?

Reply:
Yes. I'll do that.